### PR TITLE
Make README more appealing

### PR DESCRIPTION
**Pull request by chaudum**
## Summary of the changes / Why this is an improvement

Related issues: #64 #73 

## Checklist

 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: .yamllint [hunk 1]
```diff
@@ -11,5 +11,4 @@ rules:
   indentation:
     indent-sequences: consistent
     spaces: 2
-  key-ordering: enable
```
Comment (MarkusH) [inline .yamllint:14]: I'd love for us to keep sorting of keys in YAML files. Makes it much easier to find things, especially in bigger files.

File: README.rst [hunk 2]
```diff
@@ -1,15 +1,119 @@
...
+.. code-block:: console
+
+   $ kubectl create -f deploy/rbac.yaml
+   ...
+   $ kubectl create -f deploy/deployment.yaml
+   ...
+
+Please refer to the `configuration documentation`_ for further details.
+
+💻 Development
```
Comment (MarkusH) [inline README.rst:113]: I think we could move the whole development section into https://github.com/crate/crate-operator/blob/master/docs/source/development.rst instead.

**Review by MarkusH — COMMENTED** [review]
:heart:

File: .yamllint (hunk 1 above)
Comment (chaudum) [reply .yamllint:14]: This is not very practical when dealing with Kubernetes manifest YAMLs :)

File: README.rst (hunk 2 above)
Comment (chaudum) [reply README.rst:113]: Yeah, makes sense

File: .yamllint (hunk 1 above)
Comment (MarkusH) [reply .yamllint:14]: How's that? When I look at `kubectl get deployments crate-operator -o yaml`, the keys are sorted alphabetically. Exactly the way `key-ordering` enforces them.

**Comment by MarkusH** [comment]
Would you mind prefixing the commit message with `Fix #64: ` such that the issue is linked and closed automatically when the PR is merged.

File: .yamllint (hunk 1 above)
Comment (chaudum) [reply .yamllint:14]: Well, it's the same reason why `black` fails to make properly **human** readable Python code ;)

I usually aim for "easy to read for humans", e.g. compare:

```yaml
    spec:
      containers:
      - name: crate-operator
        image: "crate/crate-operator:latest"
        imagePullPolicy: Always
        env:
        - name: CRATEDB_OPERATOR_BOOTSTRAP_TIMEOUT
          value: "1800"
        - name: CRATEDB_OPERATOR_DEBUG_VOLUME_SIZE
          value: "128GiB"
        - name: CRATEDB_OPERATOR_DEBUG_VOLUME_STORAGE_CLASS
          value: "default"
        - name: CRATEDB_OPERATOR_JMX_EXPORTER_VERSION
          value: "0.6.0"
        - name: CRATEDB_OPERATOR_LOG_LEVEL
          value: "INFO"
        - name: CRATEDB_OPERATOR_ROLLING_RESTART_TIMEOUT
          value: "3600"
        - name: CRATEDB_OPERATOR_SCALING_TIMEOUT
          value: "3600"
```

vs.

```yaml
    spec:
      containers:
      - env:
        - name: CRATEDB_OPERATOR_BOOTSTRAP_TIMEOUT
          value: "1800"
        - name: CRATEDB_OPERATOR_DEBUG_VOLUME_SIZE
          value: "128GiB"
        - name: CRATEDB_OPERATOR_DEBUG_VOLUME_STORAGE_CLASS
          value: "default"
        - name: CRATEDB_OPERATOR_JMX_EXPORTER_VERSION
          value: "0.6.0"
        - name: CRATEDB_OPERATOR_LOG_LEVEL
          value: "INFO"
        - name: CRATEDB_OPERATOR_ROLLING_RESTART_TIMEOUT
          value: "3600"
        - name: CRATEDB_OPERATOR_SCALING_TIMEOUT
          value: "3600"
        image: "crate/crate-operator:latest"
        imagePullPolicy: Always
        name: crate-operator
```

Well, that's my few cents ;-)

**Comment by chaudum** [comment]
> Would you mind prefixing the commit message with `Fix #64: ` such that the issue is linked and closed automatically when the PR is merged.

Yes, I do mind. The place in the first line of a commit message is super sparse anyway and I think the most prominent real estate should not be used for referencing issues.

**Review by Taliik — COMMENTED** [review]
Really like it. Since the `Readme` now contains the `installation` steps, should we remove the dedicated `docs/source/installation.rst` installation file?

**Comment by chaudum** [comment]
> Really like it. Since the `Readme` now contains the `installation` steps, should we remove the dedicated `docs/source/installation.rst` installation file?

Docs and README are two separate things. I'd rather extend the installation docs with the information from the README.

**Comment by MarkusH** [comment]
> > Would you mind prefixing the commit message with `Fix #64: ` such that the issue is linked and closed automatically when the PR is merged.
> 
> Yes, I do mind. The place in the first line of a commit message is super sparse anyway and I think the most prominent real estate should not be used for referencing issues.
> 
> Again, that's just my point of view ;)

I'm okay with that, if you include it in the body of the commit message instead :slightly_smiling_face: Thanks!

**Review by MarkusH — APPROVED** [review]
:heavy_check_mark: pending review from @crate/tech-writing

**Comment by chaudum** [comment]
> I'm okay with that, if you include it in the body of the commit message instead slightly_smiling_face Thanks!

Yeah, will do that. :)

**Pull request merged**